src/update.ts

```diff
@@ -268,14 +268,18 @@ export class Updater {
     return current === updated
   }
 
-  private async determineChannel(): Promise<string> {
-    const channelPath = path.join(this.config.dataDir, 'channel')
-    if (fs.existsSync(channelPath)) {
-      const channel = await fs.readFile(channelPath, 'utf8')
-      return String(channel).trim()
+  private async determineChannel(version:string|undefined): Promise<string> {
+    try {
+      const {body} = await HTTP.get(`https://registry.npmjs.org/${this.config.pjson.name}/`)
```

Comment:
proxy should be ok.  That library looks to have implemented it
https://github.com/search?q=repo%3Aheroku%2Fhttp-call%20proxy&type=code